src/sources/datadog/logs.rs

```diff
@@ -77,6 +78,56 @@ struct DatadogLogsSource {
     store_api_key: bool,
 }
 
+// https://github.com/DataDog/datadog-agent/blob/a33248c2bc125920a9577af1e16f12298875a4ad/pkg/logs/processor/json.go#L23-L49
+#[derive(Deserialize, Clone, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+struct LogMsg {
+    pub message: Bytes,
+    pub status: Bytes,
+    pub timestamp: i64,
+    pub hostname: Bytes,
+    pub service: Bytes,
+    pub source: Bytes,
+    pub tags: Bytes,
+}
+
+fn decode_body(
+    body: Bytes,
+    api_key: Option<Arc<str>>,
+    events: &mut Vec<Event>,
+) -> Result<(), ErrorMessage> {
+    let messages: Vec<LogMsg> = serde_json::from_slice(&body).map_err(|error| {
+        ErrorMessage::new(
+            StatusCode::BAD_REQUEST,
+            format!("Error parsing JSON: {:?}", error),
+        )
+    })?;
+    let timestamp_key = log_schema().timestamp_key();
+    let source_type_key = log_schema().source_type_key();
+    let now = Utc::now();
+    let iter = messages
+        .into_iter()
+        .map(|msg| {
+            let mut log = LogEvent::default();
+            log.insert_flat(timestamp_key, now);
```

Comment:
What's the interaction if the `timestamp_key` == "timestamp"? Does it conflict with ln116 or are these nested separately?

Reply:
Good question. The previous behavior was to overwrite if the log schema's timestamp key was "timestamp". That behavior is preserved here too. 

What _has_ changed, it now occurs to me, is the difference in `insert` versus `insert_flat`. I suppose it's possible that the log schema would be set to insert the timestamp in some subtree which is no longer possible, related to the conversation in #8331.

Reply:
Not to add _more_ complexity but we do have precedence in the `kubernetes_logs` source to record the "ingestion" timestamp and the "event's" timestamp

Reply:
@spencergilbert I would support this change in a follow up ticket, probably part of the broader push to tidy up observability?

Reply:
Sure! Just wanted to surface it, I only recently found out it was a thing too 😄 

I generally can't think of a reason to not use the event's timestamp (unless you want to be able to measure how long it takes for events to go from writing to aggregation 🤷 )